docs/data/data-grid/performance/performance.md

```diff
@@ -2,43 +2,51 @@
 
 <p class="description">Improve the performance of the DataGrid using the recommendations from this guide.</p>
 
-## Memoize inner components with `React.memo`
+## Extract static objects and memoize root props
 
-The `DataGrid` component is composed of a central state object where all data is stored.
-When an API method is called, a prop changes, or the user interacts with the UI (e.g. filtering a column), this state object is updated with the changes made.
-To reflect the changes in the interface, the component must re-render.
-Since the state behaves like `React.useState`, the `DataGrid` component will re-render its children, including column headers, rows, and cells.
-With smaller datasets, this is not a problem for concern, but it can become a bottleneck if the number of rows increases, especially if many columns render [custom content](/x/react-data-grid/column-definition/#rendering-cells).
-One way to overcome this issue is using `React.memo` to only re-render the child components when their props have changed.
-To start using memoization, import the inner components, then pass their memoized version to the respective slots, as follow:
+The `DataGrid` component uses `React.memo` to optimize its performance, which means itself and its subcomponents only
```

Comment:
Also tried a sandbox from https://github.com/mui/mui-x/issues/569#issuecomment-1403571943
v6.5.0: https://codesandbox.io/s/loving-paper-2fhtsq?file=/src/Demo.js
This PR: https://codesandbox.io/s/great-hopper-ufcwdd?file=/src/Demo.js